sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -45,7 +45,8 @@ class SparkOptimizer(
       PartitionPruning) :+
     Batch("InjectRuntimeFilter", FixedPoint(1),
       InjectRuntimeFilter,
-      RewritePredicateSubquery) :+
+      RewritePredicateSubquery,
+      OptimizeSubqueries) :+
```

Comment:
Thanks for the great catch, @Flyangz!

Does it work if we do the following here:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala#L88

`ConstantFolding(ColumnPruning(Aggregate(Nil, Seq(alias), filterCreationSidePlan)))`

`OptimizeSubqueries` seems a bit of an overkill as
- it runs all rules again for other non-BloomFilter subqueries as well
- the BoomFilter creation query's input to the BloomFilterAggregate has already been optimized

Reply:
Great idea, done.